cql3/functions/functions.cc

```diff
@@ -359,6 +439,14 @@ functions::get(data_dictionary::database db,
         return make_from_json_function(db, keyspace, receiver->type);
     }
 
+    // FIXME: add proper support for generic functions
+    if (name.has_keyspace()
+                ? name == SET_INTERSECTION_FUNCTION_NAME
+                : name.name == SET_INTERSECTION_FUNCTION_NAME.name) {
+        return get_set_intersection_function(db, keyspace, name, provided_args, receiver_ks, receiver_cf, receiver);
```

Comment:
I couldn't figure out in this code where you handle "unknown" types (which I thought means coming from prepare statement values). Where do you check it is the same type as the "known" types?

Reply:
If you write `WHERE my_set = set_intersection({1, 2}, ?)` prepare will pass a known type and an unknown type. function::get will return a function with two known argument types, which will then be used to set the type of the bind variable.

I see it's broken due to other reasons (and also not very useful, due to lack of flexibility in function placement in the WHERE clause), so it can't be tested right now.

Reply:
So it's handled here:

```c++
    return make_native_scalar_function<true>("set_intersection", set_type, {set_type, set_type},
```

The unknown types are replaced with the unique known type.

Reply:
Oh, makes sense. Good.